UnityProject/Assets/Scripts/Items/Weapons/Projectiles/Behaviours/ProjectileMine.cs

```diff
@@ -1,15 +1,30 @@
 ﻿using UnityEngine;
+using Tiles;
 
 namespace Weapons.Projectiles.Behaviours
 {
 	/// <summary>
-	/// Mines mineable walls on collision
+	/// Mines mineable walls on collision if hardness of the projectile is equal to or exceeds the tile's hardness
 	/// </summary>
+
 	public class ProjectileMine : MonoBehaviour, IOnHitInteractTile
 	{
+
+		[Range(1, 10)]
+		[Tooltip("what degree of hardness this projectile can overcome. Higher means the projectile can mine more types of things.")]
+		[SerializeField] private int projectileHardness = 5;
+
 		public virtual bool Interact(MatrixManager.CustomPhysicsHit hit, InteractableTiles interactableTiles, Vector3 worldPosition)
```

Comment:
reading this again I think it lacks a feedback for when it doesn't meet the requirement

Reply:
Diddly done. there is text feedback and the option to play sfx when it fails.